File: Chapter 1/Section 1.3/palsquare/palsquare.py
"""
ID: ten.to.1
LANG: PYTHON3
TASK: palsquare
"""

#!/usr/bin/python

#Files
palsquarein  = open("palsquare.in", "r")
palsquareout = open("palsquare.out", "w")

#Variables
base          = int(palsquarein.read())
maxnum        = 301;

def base10tobaseX(base10num, baseX):
	valtosym = {
	0  : "0",
	1  : "1",
	2  : "2",
	3  : "3",
	4  : "4",
	5  : "5",
	6  : "6",
	7  : "7",
	8  : "8",
	9  : "9",
	10 : "A",
	11 : "B",
	12 : "C",
	13 : "D",
	14 : "E",
	15 : "F",
	16 : "G",
	17 : "H",
	18 : "I",
	19 : "J",
	20 : "K"
	}
	baseXnum = []
	
	while (base10num > 0):
		(base10num, val) = divmod(base10num, baseX)
		baseXnum.append(valtosym[val])
	
	return ''.join(list(reversed(baseXnum)))

def checkpalindrome(numlist):
	if(numlist == numlist[::-1]):
		return True
	else:
		return False

#Brute Force
for n in range(1, maxnum):
	base10num   = base10tobaseX(n, base)
	basenumsqrd = base10tobaseX(n ** 2, base)
	output      = base10num + " " + basenumsqrd + "\n"
	
	if(basenumsqrd == basenumsqrd[::-1]):
		palsquareout.write(output)

File: Chapter 1/Section 1.3/palsquare/palsquare.cpp
/*
ID: ten.to.1
LANG: C++
TASK: palsqure
*/

#include <iostream>
#include <fstream>
#include <string>
#include <vector>

//Base X <= 10
int base10tobaseX(int base10num, int baseX)
{
	int numbaseX = 0;
	int k = 1;
	
	while(base10num > 0)
	{
		numbaseX += (base10num % baseX) * k;
		k *= 10;
		base10num /= baseX;
	}
	
	return numbaseX;
}

bool checkpalindrome()
{
	
}

int main()
{
	//Files
	std::ifstream palsquarein("palsquare.in");
	std::ofstream palsquareout("palsquare.out");
	
	//Variables
	int base; palsquarein >> base;
	int maxnum = 300;
	
	//Brute Force
	for(int num; num <= maxnum; num++)
	{
		
	}
	//WHY DID I START THIS IN C++?
}

